src/makeInfinityAdder.js

```diff
@@ -4,7 +4,25 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let flag = 0;
+  let sum = 0;
+
+  const makeSum = (a) => {
+    if (typeof a === 'number') {
+      sum += a;
+      flag = 0;
+
+      return makeSum;
+    } else if (typeof a !== 'number') {
+      flag++;
+
+      sum = flag === 2 ? 0 : sum;
```

Comment:
The logic for resetting the sum to 0 when 'flag' equals 2 is incorrect according to the task requirements. The sum should be reset to 0 only after returning the result when a call is made without parameters.